lib/active_merchant/billing/gateways/pagarme.rb

```diff
@@ -228,8 +228,13 @@ def json_error(raw_response)
 
       def success_from(response)
         success = ["subscription", "plan", "transaction"]
+        status = ["paid","authorized","refunded"]
 
-        success.include? response["object"]
+        success.include?(response["object"]) && status_verify(status,response) || success.include?(response["object"])
```

Comment:
@mandado aqui você pode fazer assim. Dessa maneira você só verificar os status se for uma request que o `object` é `transaction`.

```
if success.include?(response["object"])
  response["object"] == "transaction" ? status.include?(response['status']) : true
end
```

obs: não precisa por o `response.key?('status')` porque se o `response['status']` não existir vai retornar `nil`, e como `status` é igual a `["paid","authorized","refunded"]` o `status.include?(nil)` vai te retornar `false`.